liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java

```diff
@@ -815,6 +799,43 @@ public void undeploy(final Archive<?> archive) throws DeploymentException
          log.exiting(className, "undeploy");
       }
    }
+ 
+   private void deleteWithRetries(File archveToDelete, boolean isFailSafeUndeployment, int retries, int retryInterval) throws DeploymentException {
+      int retryNo = -1;
+      boolean complete = false;
+      do {
+         if (retryNo >= 0) {
+            try {
+               Thread.sleep(retryInterval);
+            } catch (InterruptedException e) {
+               throw new DeploymentException("Interrupted Exception", e);
+            }
+         }
+         if (!containerConfiguration.isFailSafeUndeployment()) {
```

Comment:
Is the failSafeUndeployment option still needed? If so how does it interact with doing retries?

I note that the failSafeUndeployment documentation only talks about xml files (rather than application archives).

Either way, deleteOnExit isn't a great solution for application archives since some tests may re-use the same application name for different tests.

Reply:
Ok, on a second reading, I can see that we're now only doing the `deleteOnExit()` if we go through all the retries.